### ISSUE TITLE
VRAM overhead with log prob storage

### ISSUE THREAD

**Issue by entfane** [entfane/RLHF#7]
1) Log probs are calculated for the whole vocabulary of online policy
2) KL divergence is calculated per token using the whole distribution.

**Comment by entfane** [comment]
Problem is in vram usage. Incorrect release of vram. Have to look into details regarding reference logits, log probs, values, etc.
